plugins/optimization-detective/class-od-tag-visitor-context.php

```diff
@@ -46,13 +48,47 @@ final class OD_Tag_Visitor_Context {
 	/**
 	 * Constructor.
 	 *
-	 * @param OD_HTML_Tag_Processor           $processor                    HTML tag processor.
-	 * @param OD_URL_Metrics_Group_Collection $url_metrics_group_collection URL metrics group collection.
-	 * @param OD_Link_Collection              $link_collection              Link collection.
+	 * @param OD_HTML_Tag_Processor          $processor                   HTML tag processor.
+	 * @param OD_URL_Metric_Group_Collection $url_metric_group_collection URL metric group collection.
+	 * @param OD_Link_Collection             $link_collection             Link collection.
+	 */
+	public function __construct( OD_HTML_Tag_Processor $processor, OD_URL_Metric_Group_Collection $url_metric_group_collection, OD_Link_Collection $link_collection ) {
+		$this->processor                   = $processor;
+		$this->url_metric_group_collection = $url_metric_group_collection;
+		$this->link_collection             = $link_collection;
+	}
+
+	/**
+	 * Gets magic properties.
+	 *
+	 * @param string $name Property name.
+	 * @return OD_URL_Metric_Group_Collection URL metric group collection.
+	 *
+	 * @throws Error When property is unknown.
 	 */
-	public function __construct( OD_HTML_Tag_Processor $processor, OD_URL_Metrics_Group_Collection $url_metrics_group_collection, OD_Link_Collection $link_collection ) {
-		$this->processor                    = $processor;
-		$this->url_metrics_group_collection = $url_metrics_group_collection;
-		$this->link_collection              = $link_collection;
+	public function __get( string $name ): OD_URL_Metric_Group_Collection {
```

Comment:
This is needed to account for someone updating Optimization Detective but not right away updating Image Prioritizer or Embed Optimizer.  They are using the `url_metrics_group_collection` symbol, so without this back-compat code, a fatal error would occur. This is similar to following code in Image Prioritizer:

https://github.com/WordPress/performance/blob/5952d556d9632f93c2fbbda76c3516576ebdb1a5/plugins/image-prioritizer/class-image-prioritizer-img-tag-visitor.php#L69-L88

The `is_any_group_populated()` method was added to Optimization Detective in a recent release, and to avoid a fatal error if someone updates Optimization Detective without updating Image Prioritizer as well, this prevents the code from causing a fatal error.

All of this code could be removed later once there is no longer this risk.

Reply:
Let's add a `TODO` to remove this later.

Reply:
Added in 5960ff3c